cpp/include/typedb/concept/thing/attribute.hpp

```diff
@@ -0,0 +1,43 @@
+
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+#include "typedb/concept/thing/thing.hpp"
+#include "typedb/concept/value/value.hpp"
+
+namespace TypeDB {
+
+class Concept;
+
+class Attribute : public Thing {
+    friend class Concept;
+
+   private:
+    std::optional<Value> value;
```

Comment:
Should I store it at all? Or just hit the native layer in the function?
It ties into the question of whether it's the same native pointer each time.